python/lsst/ts/hexrotcomm/base_mock_controller.py

```diff
@@ -295,3 +321,138 @@ async def update_telemetry(self, curr_tai):
+
+    async def close_client(self):
+        self.command_loop_task.cancel()
+        self.telemetry_loop_task.cancel()
+        await super().close_client()
+
+    async def telemetry_loop(self):
+        """Write configuration once, then telemetry at regular intervals."""
+        self.log.info("telemetry_loop begins")
+        try:
+            if self.connected:
+                await self.write_config()
+            # Checking self.writer is not None makes mypy happy
+            while self.connected and self.writer is not None:
+                header, curr_tai = self.update_and_get_header(self.telemetry.FRAME_ID)
+                await self.update_telemetry(curr_tai=curr_tai)
+                async with self.write_lock:
+                    await tcpip.write_from(self.writer, header, self.telemetry)
+                await asyncio.sleep(self.telemetry_interval)
+            self.log.info("Socket disconnected")
+        except asyncio.CancelledError:
+            raise
+        except Exception:
+            self.log.exception("telemetry_loop failed")
+
+    def update_and_get_header(self, frame_id):
+        """Update the config or telemetry header and return it and the time.
+
+        Call this prior to writing telemetry or configuration.
+
+        Parameters
+        ----------
+        frame_id : `int`
+            Frame ID of header to write.
+
+        Returns
+        -------
+        header : `structs.Header`
+            The header.
+        curr_tai : `float`
+            Current time in header timestamp (TAI, unix seconds).
+        """
+        header = self.headers[frame_id]
+        curr_tai = utils.current_tai()
+        tai_frac, tai_sec = math.modf(curr_tai)
+        header.tai_sec = int(tai_sec)
+        header.tai_nsec = int(tai_frac * 1e9)
+        return header, curr_tai
+
+    async def write_config(self):
+        """Write the current configuration."""
+        # Checking self.writer is not None makes mypy happy
+        if not self.connected or self.writer is None:
+            raise RuntimeError("Not connected")
+        header, curr_tai = self.update_and_get_header(self.config.FRAME_ID)
+        async with self.write_lock:
+            await tcpip.write_from(self.writer, header, self.config)
+
+    async def write_command_status(self, counter, status, duration=0, reason=""):
+        """Write a command status.
+
+        Parameters
+        ----------
+        counter : `int`
+            counter field of command being acknowledged.
+        status : `CommandStatusCode`
+            Command status code.
+        duration : `float` or `None`, optional
+            Estimated duration. None is treated as 0.
+        reason : `str`, optional
+            Reason for failure. Should be non-blank if and only if the
+            command failed.
+        """
+        # Checking self.writer is not None makes mypy happy
+        if not self.connected or self.writer is None:
+            raise RuntimeError("Not connected")
```

Comment:
Please consider to add the doc string of `RuntimeError`.

Reply:
I don't understand what you are asking for.